src/PrestaShopBundle/Controller/Admin/Improve/International/CurrencyController.php

```diff
@@ -388,6 +388,8 @@ public function refreshExchangeRatesAction()
     /**
      * Handles ajax request which updates live exchange rates.
      *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller')) || is_granted('delete', request.get('_legacy_controller'))")
```

Comment:
Here as well you could clean the manual check in the controller

Reply:
No it's another case where message "Access denied" must be in JSON